src/docs/product/integrations/integration-platform/public-integration.mdx

```diff
@@ -0,0 +1,151 @@
+
+### Refreshing Tokens
+
+In the initial installation, you'll need the grant code given to you in either the installation webhook request or the redirect URL, in addition to your integration's client ID and client Secret:
+
+```python
+url = u'https://sentry.io/api/0/sentry-app-installations/{}/authorizations/'
+url = url.format(install_id)
+
+payload = {
+    'grant_type': 'authorization_code',
+    'code': code,
+    'client_id': 'your-client-id',
+    'client_secret': 'your-client-secret',
+}
+```
+
+Tokens expire after eight hours, so you'll need to refresh your tokens accordingly:
+
+```python
+def refresh_token(install_id):
+    url = u'https://sentry.io/api/0/sentry-app-installations/{}/authorizations/'
+    url = url.format(install_id)
+
+    refresh_token = retrieve_refresh_token_from_db(install_id)
+
+    payload = {
+        'grant_type': 'refresh_token',
+        'refresh_token': refresh_token,
+        'client_id': 'your-client-id',
+        'client_secret': 'your-client-secret',
+    }
+
+    resp = requests.post(url, json=payload)
+    data = resp.json()
+
+    new_token = data['token']
+    new_refresh_token = data['refreshToken']
+    # ... Securely update the token and refresh_token in DB...
+
+    return new_token
+```
+
+The data you can expect back for both the initial grant code exchange and subsequent token refreshes is as follows:
+
+```json
+{
+  "id": "38",
+  "token": "ec48bf98637d44c294ead7566513686237e74ab67a074c64b3aaca2d93dbb8f1",
+  "refreshToken": "c866f154a65841638d44ee26364409b0a1a67bd642bd46e7a476f34f810712d6",
+  "dateCreated": "2019-08-07T20:25:09.870Z",
+  "expiresAt": "2019-08-08T04:25:09.870Z",
+  "state": null,
+  "application": null
+}
+```
+
+## Verifying Installations
+
+Typically, if you have the redirect URL configured, there is work happening on your end to "finalize" the installation. If this is the case, we recommend enabling the "Verify Install" option for your integration. Once enabled, you'll need to send a request marking the installation as officially "installed":
+
+```python
+requests.put(
+    u'https://sentry.io/api/0/sentry-app-installations/{}/'.format(install_id),
+    json={'status': 'installed'},
+)
+```
+
+## Uninstallation
+
+When a user uninstalls your integration, you'll receive an `installation.deleted` request to your webhook URL. For more information, check out the [full Webhooks documentation](/product/integrations/integration-platform/webhooks/).
+
+## Authorized Origins
+
+It's possible to use auth tokens from the browser if you allow the origins of the pages making the requests. In the "Authorized JavaScript Origins" field, add each origin (for example, docs.sentry.io) separated by a newline. You don't need to include the protocol (http or https) in the origin. Currently, you can't use any wildcard characters (for example, `*.sentry.io`), so if you have multiple subdomains, you'll need to add them individually.
+
+## Image Requirements
```

Comment:
Both of the sections (internal and public) for Image Requirements have this weird formatting thing where the first three bullets have no newline spaces after them, but the second list has a space between.

Not sure which we like better but I think we should stick to one style for them both.
![image](https://user-images.githubusercontent.com/35509934/167730707-f53989f1-4dc5-464d-a8b3-318b09547bf4.png)